doc/users/fonts.rst

```diff
@@ -0,0 +1,120 @@
+All about Matplotlib and Fonts!
+===============================
+
+The story of fonts has been quite eventful throughout time. It involves
+contributions of tech giants such as the likes of Adobe, Apple and Microsoft.
+
+Types
+-----
+In practice, there are 3 types Matplotlib supports (in addition to
+'core fonts', more about which is explained later in the guide):
+
+.. list-table:: Type of Fonts
+   :header-rows: 1
+
+   * - Type 1 (PDF)
+     - Type 3 (PDF/PS)
+     - TrueType (PDF)
+   * - One of the oldest types, introduced by Adobe
+     - Similar to Type 1 in terms of introduction
+     - Newer than previous types, used commonly today, introduced by Apple
+   * - Restricted subset of PostScript, charstrings are in bytecode
+     - Full PostScript language, allows embedding arbitrary code
+       (in theory, even render fractals when rasterizing!)
+     - Include a virtual machine that can execute code!
+   * - These fonts support font hinting
+     - Do not support font hinting
+     - Hinting supported (virtual machine processes the "hints")
+   * - Non-subsetted through Matplotlib
+     - Subsetted via external module `ttconv <https://github.com/sandflow/ttconv>`_
+     - Subsetted via external module `fonttools <https://github.com/fonttools/fonttools>`_
+
+NOTE: Adobe will disable support for authoring with Type 1 fonts in
+January 2023. `Read more here. <https://helpx.adobe.com/fonts/kb/postscript-type-1-fonts-end-of-support.html>`_
+
+Special Mentions
+~~~~~~~~~~~~~~~~
+- Type 42 fonts (PS):
+
+  - PostScript wrapper around TrueType fonts
+  - 42 is the `Answer to Life, the Universe, and Everything! <https://en.wikipedia.org/wiki/Answer_to_Life,_the_Universe,_and_Everything>`_
+  - Matplotlib uses an external library called `fonttools <https://github.com/fonttools/fonttools>`_
+    to subset these types of fonts
+
+- OpenType fonts:
+
+  - OpenType is a new standard for digital type fonts, developed jointly by
+    Adobe and Microsoft
+  - Generally contain a much larger character set!
+  - Limited Support with Matplotlib
+
+
+Subsetting
+----------
+Matplotlib is able to generate documents in multiple different formats. Some of
+those formats (for example, PDF, PS/EPS, SVG) allow embedding font data in such
+a way that when these documents are visually scaled, the text does not appear
+pixelated.
+
+This can be achieved by virtually embedding the *whole* font file within the
+output document. However, this can lead to **very large documents**, wherein
+most of the size bandwidth is captured by that font file data.
+
+Font Subsetting is a way to embed only the *required* glyphs within the
```

Comment:
```suggestion
Font subsetting embeds only the *required* glyphs within
```

Reply:
Actually, subsetting doesn't equate to embedding. We embed _after_ we subset, so 'font subsetting' is a 'way' to embed...
(maybe "Font Subsetting can be used to embed...")

Reply:
Are there other ways to embed only the required glyphs, other than subsetting?  If not, then the phrase "is a way" is needlessly indirect and mysterious.  Perhaps you are trying to make a distinction here I dont understand?

Reply:
Actually there's no connection of 'embedding' with 'subsetting', and I tried to make exactly that distinction.

_With_ subsetting, the no. of glyphs are reduced, but at no point does it interfere with the 'embedding' part. Embedding is that dumb stage in the pipeline at which we can throw either a subsetted, or a non-subsetted font, and it will embed it in the document happily.

Reply:
I've anyway changed it to:
```
Font Subsetting can be used before generating documents, to embed only the
*required* glyphs within the documents.
```